### PR TITLE
Update OrderItemValidator to require a delivery date on Additional Services

### PR DESCRIPTION
**Pull request by leerogers-bjss**
Update CreateOrderItemModelValidator to check for delivery date on Additional Services also.

And update tests and add new tests to reflect this

Story: [AB#9272](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/9272)
Task: [AB#13546](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13546)

### CODE REVIEW

**Review by PhysicsJon — APPROVED** [review]
Looks good, probably needs a brief look by @bjss-chris

**Review by bjss-chris — CHANGES_REQUESTED** [review]
I think we need to add a unit test to cover the validation for an additional service with a missing delivery date, unless I've missed it?

We can probably update `Validate_DeliveryDateIsNull_IsSolution_HasError` (with a rename) to use inline data that tests both solutions and additional services.

Looks good on the whole though.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Validation/CreateOrderItemModelValidatorTests.cs [hunk 1]
```diff
@@ -421,9 +421,28 @@ static CreateOrderItemModelValidatorTests()
...
+                $"{nameof(CreateOrderItemModel.ServiceRecipients)}[0].{nameof(OrderItemRecipientModel.DeliveryDate)}";
+
+            var result = validator.Validate(model);
+            var failure = result.Errors.FirstOrDefault(e => e.PropertyName == expectedPropertyName);
+
+            failure.Should().BeNull();
+        }
+
+        [Test]
         [CommonInlineAutoData(nameof(CatalogueItemType.AssociatedService))]
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Validation/CreateOrderItemModelValidatorTests.cs:444]: It's probably not worth using inline data here anymore as we only have the one test case now.

Comment (leerogers-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Validation/CreateOrderItemModelValidatorTests.cs]: I've Added in the IsNull_HasError test and updated the Associated service one to use AutoData not inline now

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Validation/CreateOrderItemModelValidatorTests.cs [hunk 2]
```diff
@@ -421,15 +421,54 @@ static CreateOrderItemModelValidatorTests()
...
+
+            var expectedPropertyName =
+                $"{nameof(CreateOrderItemModel.ServiceRecipients)}[0].{nameof(OrderItemRecipientModel.DeliveryDate)}";
+
+            var result = validator.Validate(model);
+            var failure = result.Errors.FirstOrDefault(e => e.PropertyName == expectedPropertyName);
+
+            Assert.NotNull(failure);
+            failure.ErrorMessage.Should().Be($"{nameof(OrderItemRecipientModel.DeliveryDate)}Required");
+        }
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Validation/CreateOrderItemModelValidatorTests.cs:461]: Can we refactor this test and `Validate_DeliveryDateIsNull_IsSolution_HasError` to a single inline data test based on the catalogue item type?

Comment (leerogers-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Validation/CreateOrderItemModelValidatorTests.cs:461]: should be able too. will do it now

**Review by bjss-chris — APPROVED** [review]

**Comment by bjss-chris** [comment]
@leerogers-bjss – branch just needs bringing up-to-date and you can merge this.

**Pull request merged**